### PR TITLE
Configuration management

### PR DESCRIPTION
**Pull request by benichu**
@mdeloupy : ready to review.

This implements the configuration setup and save mechanism, for example, executing the command `$ checklister setup` gives you this kind of configuration wizard:

You can also read the currently saved services by using this command `$ checklister settings` after picking the services you are interested in, it will display the currently saved configuration (by default, it's in a file `~/.checklister.yml`

Once this branch is ready to squash and merge, we should have an easy way to access user defined settings in our gem (especially when interacting with gitlab remote service...)

### CODE REVIEW

File: lib/checklister/configuration.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
+module Checklister
```
Comment (benichu) [inline lib/checklister/configuration.rb]: @mdeloupy can you make sense of that documentation, it's a draft to help me clearly define how we set, store and retrieve configuration values. We need to take into consideration that one configuration key (ex: `username`) can have several values stored (one for github or for gitlab, etc...). And that every time a user executes a `checklister` command,  a context is selected and the proper value can be used transparently by the program...

Comment (benichu) [inline lib/checklister/configuration.rb]: I advise use to read the documentation via the web interface at [http://localhost:8808/docs/Checklister/Configuration](http://localhost:8808/docs/Checklister/Configuration) after running the `script/doc` command.

Comment (mdeloupy) [inline lib/checklister/configuration.rb]: I don't understand, what do you want me to do ?

Comment (benichu) [inline lib/checklister/configuration.rb]: Read, understand, comment...

Comment (benichu) [inline lib/checklister/configuration.rb]: Ask me about anything that is not clearly explained.

Comment (mdeloupy) [inline lib/checklister/configuration.rb]: Be critical, I can do that :-)

**Comment by benichu** [comment]
For information, it's missing some documentation and things should be improved/refactored, but I'd like to plan that for later in the development process and have a beta application first.

**Pull request merged**